private/bufpkg/bufconfig/write.go

```diff
@@ -273,30 +257,162 @@ func writeConfig(
 	for _, option := range options {
 		option(writeConfigOptions)
 	}
-	if !writeConfigOptions.documentationComments && writeConfigOptions.uncomment {
-		return errors.New("cannot set uncomment without documentationComments for WriteConfig")
+	if err := validateWriteConfigOptions(writeConfigOptions); err != nil {
+		return err
 	}
-	if writeConfigOptions.moduleIdentity == nil && len(writeConfigOptions.dependencyModuleReferences) > 0 {
-		return errors.New("cannot set deps without a name for WriteConfig")
+	// This is the same default as the bufconfig getters.
+	version := V1Beta1Version
+	if writeConfigOptions.version != "" {
+		if err := validateVersion(writeConfigOptions.version); err != nil {
+			return err
+		}
+		version = writeConfigOptions.version
 	}
-	externalConfigV1 := ExternalConfigV1{
-		Version: V1Version,
+	config := &Config{
+		Version:        version,
+		ModuleIdentity: writeConfigOptions.moduleIdentity,
 	}
-	externalConfigV1.Lint.Use = defaultLintIDs
-	externalConfigV1.Breaking.Use = defaultBreakingIDs
-	if writeConfigOptions.moduleIdentity != nil {
-		externalConfigV1.Name = writeConfigOptions.moduleIdentity.IdentityString()
+	if writeConfigOptions.breakingConfig != nil {
+		config.Breaking = writeConfigOptions.breakingConfig
+		// Always force the provided version.
+		config.Breaking.Version = version
 	}
-	for _, dependencyModuleReference := range writeConfigOptions.dependencyModuleReferences {
-		externalConfigV1.Deps = append(
-			externalConfigV1.Deps,
-			dependencyModuleReference.String(),
-		)
+	if writeConfigOptions.lintConfig != nil {
+		config.Lint = writeConfigOptions.lintConfig
+		// Always force the provided version.
+		config.Lint.Version = version
```

Comment:
We should validate that the provided `lintConfig.Version` is equivalent to the `version` (same with `breakingConfig` above).

Forcing the override might lead to invalid results if the configured rules changed between `v1beta` and `v1`.

Reply:
Yeah -- that's fair. The validation happens in the module build also well, but maybe it makes sense to actually validate the versions here and then move the version inferences out of `bufmodule.go`?